### PR TITLE
Use `assumeThat` to GCP/Azure ITs that need secrets

### PR DESCRIPTION
**Pull request by snazy**
This change lets the tests that require secrets not pass but mark those as "skipped".

Also a nit-ish change (suppress-warning).

### CODE REVIEW

**Comment by snazy** [comment]
@collado-mike can you check whether this still works as expected with the required secrets?

File: polaris-core/src/test/java/io/polaris/service/storage/gcp/GcpCredentialsStorageIntegrationTest.java [hunk 1]
```diff
@@ -81,57 +75,60 @@ public void testSubscope(boolean allowedListAction) throws IOException {
...
+      final byte[] fileContent = "hello-polaris".getBytes();
+      // GOOD WRITE
+      Assertions.assertThatNoException()
+          .isThrownBy(() -> storageClient.create(blobInfoGoodWrite, fileContent));
 
-    Assertions.assertThatNoException()
-        .isThrownBy(() -> storageClient.get(blobInfoGoodRead.getBlobId()));
-    Assertions.assertThatThrownBy(() -> storageClient.get(blobInfoBad.getBlobId()))
-        .isInstanceOf(StorageException.class);
+      // BAD WROTE
```
Comment (adutra) [inline polaris-core/src/test/java/io/polaris/service/storage/gcp/GcpCredentialsStorageIntegrationTest.java:90]:
```suggestion
      // BAD WRITE
```

**Review by dimas-b — APPROVED** [review]

File: polaris-core/src/test/java/io/polaris/service/storage/azure/AzureCredentialStorageIntegrationTest.java [hunk 2]
```diff
@@ -49,31 +49,25 @@
...
   private final String clientSecret = System.getenv("AZURE_CLIENT_SECRET");
   private final String tenantId = "d479c7c9-2632-445a-b22d-7c19e68774f6";
 
-  private boolean checkEnvNullVariables() {
-    if (Strings.isNullOrEmpty(clientId) || Strings.isNullOrEmpty(clientSecret)) {
-      LOGGER.debug("Null Azure testing environment variables! Skip {}", this.getClass().getName());
-      return true;
-    }
-    return false;
+  private void checkEnvNullVariables() {
```
Comment (RussellSpitzer) [inline polaris-core/src/test/java/io/polaris/service/storage/azure/AzureCredentialStorageIntegrationTest.java:67]: nit: now that this is void could be assumeEnvVariablesNotNull.

File: polaris-core/src/test/java/io/polaris/service/storage/azure/AzureCredentialStorageIntegrationTest.java [hunk 3]
```diff
@@ -250,10 +243,10 @@ public void testGetSubscopedTokenRead(boolean allowListAction, String service) {
   }
 
   @TestWithAzureArgs
-  public void testGetSubscopedTokenWrite(boolean allowListAction, String service) {
-    if (checkEnvNullVariables()) {
-      return;
-    }
+  public void testGetSubscopedTokenWrite(
+      @SuppressWarnings("unused") boolean allowListAction, String service) {
```
Comment (RussellSpitzer) [inline polaris-core/src/test/java/io/polaris/service/storage/azure/AzureCredentialStorageIntegrationTest.java:247]: why keep the parameter if we don't need it?

**Review by RussellSpitzer — DISMISSED** [review]
I think we should change the method name but I think you should feel free to merge after that.

File: polaris-core/src/test/java/io/polaris/service/storage/azure/AzureCredentialStorageIntegrationTest.java (hunk 3 above)
Comment (snazy) [reply polaris-core/src/test/java/io/polaris/service/storage/azure/AzureCredentialStorageIntegrationTest.java:247]: it's a parameterized test

**Review by RussellSpitzer — APPROVED** [review]

**Comment by RussellSpitzer** [comment]
Thanks @snazy !

**Pull request merged**